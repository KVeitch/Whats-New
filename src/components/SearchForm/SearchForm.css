header {
  grid-area: top;
  height: 80px;
  display:flex;
  align-items: center;
  color:white;
  font-size: 20px;
}

.search-bar h1 {
  display:inline-block;
}

.app-name {
  font-size: 35px;
  min-width: 250px;
}

.yellow {
  color: orange;
}

.search__input {
  height: 38px;
  width: calc(100% - 450px);
  padding-left:10px;
  border: 0px;
  background-color: #5d89b1;
  margin-left: 15px;
  font-size: 20px;
  color: white;
}

.search__input::placeholder {
  color:white;
}

.search__btn {
  height: 38px;
  background-color: #8c7ae6;
  border-radius: 19px;
  width: 180px;
  margin: 0 10px;
  line-height: 38px;
  font-size: 20px;
  cursor: pointer;
}