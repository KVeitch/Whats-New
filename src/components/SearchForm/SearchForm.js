import React, { Component } from 'react'
import './SearchForm.css';

class SearchForm extends Component {
  constructor() {
    super();
    this.state = {
      search:''
    }
  }
  handleChange = (event) => {
    this.setState({search:event.target.value});
  }

  handleSubmit = () => {
    const searchText = this.state.search.toLowerCase();
    this.props.searchNews(searchText);
    this.setState({ search:'' });
  }

  handleEnterKey = (event) => {
    if (event.keyCode === 13) {
      this.handleSubmit()
    }
  }

  render() {
    return <header className='header'>
      <h1 className='app-name'>What's <span className='yellow'>News</span>?</h1>
      <input 
        type= 'text'
        name= 'search'
        value= {this.state.search}
        onChange= {this.handleChange}
        onKeyDown= {this.handleEnterKey}
        className='search__input' 
        placeholder='Search for news articles here...'
      />
      <div 
      className='search__btn'
      onClick= {this.handleSubmit}
      >
        Search Now</div>
    </header>
  }
 }

export default SearchForm;