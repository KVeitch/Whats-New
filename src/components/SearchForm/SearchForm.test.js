import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';


describe ('SearchForm', () => {
  let wrapper;
  const mockSearchNews = jest.fn()

  beforeEach(()=> {
    wrapper = shallow(<SearchForm searchNews={mockSearchNews}/>)
  });

  it('should match snapshot', () =>{
    expect(wrapper).toMatchSnapshot()
  });
  
  describe( 'handleChange', () =>{
    it('should update state when handleChange is called', ()=>{
      let mockEvent = {
                target:{
                        value:'g'
                      }
              };
      wrapper.find('[name="search"]').simulate('onChange')
      wrapper.instance().handleChange(mockEvent)

      expect(wrapper.state().search).toEqual('g')
    });
  });

  it('Should call searchNews and clear state when the button is clicked', () => {
    let mockEvent = {
              target:{
                      value:'g'
                    }
            };
    wrapper.instance().handleChange(mockEvent)
    expect(wrapper.state().search).toEqual('g')

    wrapper.find('.search__btn').simulate('click')
    expect(wrapper.instance().props.searchNews).toHaveBeenCalled()
    expect(wrapper.state().search).toEqual('')
  });

});