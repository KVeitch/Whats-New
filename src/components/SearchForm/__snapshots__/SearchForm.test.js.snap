// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match snapshot 1`] = `
<header
  className="header"
>
  <h1
    className="app-name"
  >
    What's 
    <span
      className="yellow"
    >
      News
    </span>
    ?
  </h1>
  <input
    className="search__input"
    name="search"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Search for news articles here..."
    type="text"
    value=""
  />
  <div
    className="search__btn"
    onClick={[Function]}
  >
    Search Now
  </div>
</header>
`;
